### ISSUE TITLE
create-spark-cluster complains about virtualization type mismatch between image and instance

### ISSUE THREAD

**Issue by beaunorgeot** [BD2KGenomics/cgcloud#1]
cgcloud seems expect only pv for instance types (m3) that support both pv & hvm. Cluster creation fails w/error:
RuntimeError: Expected virtualization type paravirtual but image only supports hvm

**Comment by hannes-ucsc** [comment]
To reproduce: Run `create spark-box` followed by `create-spark-cluster -t c3.large`. The latter fails with

```
(sparkCloud)eduroam-169-233-199-178:ec2cloud bnorgeot$ cgcloud create-spark-cluster -s 2 -t c3.large
INFO: Using zone 'us-west-2a' and namespace '/bnorgeot/'
INFO: === Launching master ===
Traceback (most recent call last):
  File "/Users/bnorgeot/ec2cloud/sparkCloud/bin/cgcloud", line 9, in <module>
    load_entry_point('cgcloud-core==1.0.dev2', 'console_scripts', 'cgcloud')()
  File "/Users/bnorgeot/ec2cloud/sparkCloud/lib/python2.7/site-packages/cgcloud/core/ui.py", line 25, in main
    app.run( args )
  File "/Users/bnorgeot/ec2cloud/sparkCloud/lib/python2.7/site-packages/cgcloud/lib/util.py", line 201, in run
    command.run( options )
  File "/Users/bnorgeot/ec2cloud/sparkCloud/lib/python2.7/site-packages/cgcloud/core/commands.py", line 77, in run
    return self.run_in_ctx( options, ctx )
  File "/Users/bnorgeot/ec2cloud/sparkCloud/lib/python2.7/site-packages/cgcloud/spark/spark_cluster.py", line 46, in run_in_ctx
    ebs_volume_size=options.ebs_volume_size ) )
  File "/Users/bnorgeot/ec2cloud/sparkCloud/lib/python2.7/site-packages/cgcloud/core/commands.py", line 394, in run_on_box
    box.terminate( wait=False )
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/Users/bnorgeot/ec2cloud/sparkCloud/lib/python2.7/site-packages/cgcloud/core/commands.py", line 388, in run_on_box
    **self.instance_options( options ) )
  File "/Users/bnorgeot/ec2cloud/sparkCloud/lib/python2.7/site-packages/cgcloud/spark/spark_box.py", line 481, in prepare
    return super( SparkBox, self ).prepare( *args, **kwargs )
  File "/Users/bnorgeot/ec2cloud/sparkCloud/lib/python2.7/site-packages/cgcloud/core/box.py", line 334, in prepare
    image.virtualization_type ) )
RuntimeError: Expected virtualization type paravirtual but image only supports hvm
```

Workaround: Specify `--virtualization-type hvm` when running `create-spark-cluster`

**Comment by cket** [comment]
Closed via #34
